duckduckgo/Android
# Encourage users to add widget to home screen

Task/Issue URL: https://app.asana.com/0/361428290920652/927004026451485

**Description**:
Encourages users to add the widget to their home screen widget. Includes
- Updated CTA modal
- Add widget instructions for users whose devices do not support auto widgets / widget pinning
- A/B tests & pixels

**Steps to test this PR**:

**TEST PREPARATION:**
1. In VariantManager, comment out all variants other than "mo".
1. Open Charles and prepare to watch networking traffic

**TEST 1: CTA with option to auto add / pin from the app shown in Oreo+**
1. Run the app on an Oreo device
1. Note that the CTA is shown
1. Tap the `Install Now` button and note that the widget can be pinned to the home screen
1. Add the widget to the home screen
1. Note that the CTA is now gone
1. Remove the widget
1. Note that the CTA is back
1. Dismiss the CTA by tapping `No Thanks` and note that it does not come back again
1. Check Charles to ensure all the pixels for showing, launching, dismissing, adding and deleting the CTA were fired. See https://app.asana.com/0/361428290920652/927004026451480

**TEST 2: CTA with instructions shown on Marshmallow or Nougat**
1. Run the app on an Marshmallow device
1. Note that the CTA is shown
1. Tap the `Show Me` button and note that instructions are shown telling the user how to add the widget
1. Add the widget to the home screen
1. Dismiss the instructions and note that the CTA is now gone
1. Remove the widget
1. Note that the CTA is back
1. Dismiss the CTA by tapping `No Thanks` and note that it does not come back again
1. Check Charles to ensure all the pixels for showing, launching, dismissing, adding and deleting the CTA were fired. See https://app.asana.com/0/361428290920652/927004026451480

**TEST 3: CTA not shown while survey active**
1. Run the app cleanly
1. Intercept the survey request to https://staticcdn.duckduckgo.com/survey/survey-mobile.json and replace with and note that the survey CTA is shown
```
{
  "id": "509c0d52-f4a8-4409-a9b1-d858af42fa01",
  "surveyOptions": [
    {
      "url": "https://www.surveymonkey.com/r/9JTMFGJ",
      "installationDay": 0,
      "ratioOfUsersToShow": 1
    }
  ]
}
```

**TEST 4: CTA not shown on Lollipop devices**
1. Run the app on an Lollipop device
1. Note that the CTA is shown

**TEST 5: App Migration**
1. Install develop version of the app and run it
1. Install this branch over the top and ensure nothing goes boom

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
